examples/lessons/09/app.js

```diff
@@ -1,82 +1,100 @@
 /* eslint-disable no-var, max-statements */
 /* eslint-disable array-bracket-spacing, no-multi-spaces */
 /* global document */
-import {AnimationLoop, loadTextures, PerspectiveCamera, Group, addEvents} from 'luma.gl';
+import {GL, AnimationLoop, loadTextures, addEvents, Matrix4} from 'luma.gl';
 import {Star} from './star';
 
-var tStar;
-var twinkle = $id('twinkle');
-
 var zoom = -15;
 var tilt = 90;
 
 const animationLoop = new AnimationLoop({
-  // var canvas = document.getElementById('lesson09-canvas');
-  // canvas.width = canvas.clientWidth;
-  // canvas.height = canvas.clientHeight;
-  // var gl = createGLContext({canvas});
+  onInitialize: ({canvas, gl}) => {
+    addControls();
+    addKeyboardHandler(canvas);
 
-  onInitialize: ({gl, canvas}) => {
+    gl.clearColor(0, 0, 0, 1);
+    gl.clearDepth(1);
     gl.blendFunc(gl.SRC_ALPHA, gl.ONE);
     gl.enable(gl.BLEND);
+//    gl.disable(gl.DEPTH_TEST);
 
-    var scene = new Group(gl);
-
-    addEvents(canvas, {
-      onKeyDown(e) {
-        switch (e.key) {
-        case 'up':
-          tilt -= 1.5;
-          break;
-        case 'down':
-          tilt += 1.5;
-          break;
-        // handle page up/down
-        default:
-          if (e.code === 33) {
-            zoom -= 0.1;
-          } else if (e.code === 34) {
-            zoom += 0.1;
-          }
-        }
-      }
-    });
-
-    loadTextures(gl, {
-      urls: ['star.gif'],
-      mipmap: true,
-      parameters: [{
-        [gl.TEXTURE_MAG_FILTER]: gl.LINEAR,
-        [gl.TEXTURE_MIN_FILTER]: gl.LINEAR_MIPMAP_NEAREST
-      }]
+    return loadTextures(gl, {
+      urls: ['star.gif']
     })
     .then(textures => {
-      tStar = textures[0];
-
-      // Load all world objects
+      var stars = [];
       var numStars = 50;
       for (var i = 0; i < numStars; i++) {
-        scene.add(new Star(i / numStars * 5.0, i / numStars));
+        stars.push(new Star({
```

Comment:
Just realized that this is creating 50 copies of the program and shaders. Do you see any startup delay? 
If so the shader cache might be a worth while introduction here.

It might also be worth deleting these in `onFinalize`